common/src/main/java/com/wynntils/utils/mc/PacketUtils.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils.mc;
+
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.syncher.SynchedEntityData;
+
+public final class PacketUtils {
+
+    public static String getNameFromMetadata(List<SynchedEntityData.DataValue<?>> data) {
+        // The rename stuff we're looking for is eventually something like
+        // Optional[literal{§c§l3:46}[style={}]]
+        // or
+        // Optional[literal{§c26s}[style={}]]
+        for (SynchedEntityData.DataValue<?> packedItem : data) {
```

Comment:
This looks very much like the code that is done by the LabelHandler in @kristofbolyai's PR:
 
https://github.com/Wynntils/Artemis/pull/1146/files/8a20fe33f8e351421d4133432508810a4637e9be#diff-1d0f8b11a9ce3165c2c87c5c2bad669a7c51ef955e4d5a6e647eb027f6f8baae

I think you are both trying to solve the same problem, so we should find a unified way of doing this. Maybe if bolyai merges his PR first, and then you can see if you can use the LabelHandler as it is, or if it needs adapting to your needs?

Reply:
I got a bit carried away and created https://github.com/Wynntils/Artemis/pull/1150.

If we merge that first, you should be able to use the extended `EntityLabelChangedEvent` here as well.